src/js/model/number/float.js

```diff
@@ -0,0 +1,99 @@
+var AMPParse = AMPParse || {};
+
+
+AMPParse.buildFloatNumber = function (hexadecimal, precision)
+{
+    // Defensive programming
+    if (typeof hexadecimal === "undefined" || typeof hexadecimal !== "string")
+    {
+        throw new ReferenceError("Provided hexadecimal parameter does not exist or is not a string");
+    }
+
+    var nibbles = 0;
+
+    if (precision === "32") {
+        nibbles = 8; // 32 bits
+    } else if (precision === "64") {
+        nibbles = 16; // 64 bits
+    } else {
+        throw new ReferenceError("Provided precision is invalid: " + precision);
+    }
+
+    if (hexadecimal.length < nibbles)
+    {
+        throw new RangeError("The provided hex is too short to contain a " + precision + ".");
```

Comment:
I'd reword this to say "The provided hex is too short to contain a {{precision}} bit float."